environment/src/main/java/com/sequenceiq/environment/environment/v1/EnvironmentController.java

```diff
@@ -306,4 +311,10 @@ public Object getCreateEnvironmentForCli(EnvironmentRequest environmentRequest)
         Credential credential = credentialService.getByNameForAccountId(environmentRequest.getCredentialName(), accountId, ENVIRONMENT);
         return environmentService.getCreateEnvironmentForCli(environmentRequest, credential.getCloudPlatform());
     }
+
+    @Override
+    @DisableCheckPermissions
```

Comment:
are you sure about this?

Reply:
Yes.   This is going to eventually be an automated event that happens that will have no actual end-user initiating it so it won't have someone other than "internal actor" permissions to check against.   Validation of the manual repair action is happening at the environment API endpoint.  This API call will not be exposed as a public API and will not have an issue of external calls being made to it.

Reply:
you can authorize this similar to changeTelemetryFeatures method

```
@CheckPermissionByResourceCrn(action = AuthorizationResourceAction.EDIT_ENVIRONMENT)
    public DetailedEnvironmentResponse changeTelemetryFeaturesByEnvironmentCrn(@ResourceCrn String crn, @Valid FeaturesRequest request)
```
unless you have a good reason for it, please do not use `@DisabledCheckPermissions`

Reply:
As I said above, this is going to be an automatically called function that will not have an external user.  Do we have Internal Actor support that will allow it to have permissions checked against?

Reply:
What do you mean under automatically called event? Will it be called in flow? If you are not calling it with internal actor explicitly, in threadlocal we will find the an user Crn.

regardless of this is an automatically called method, if someone make this API call manually, they can call it withouth authz using DisabledChekPermissions annotation.

Reply:
but you can check readme of authorization-common module to find out how you can make it possible to call with internal actor

Reply:
So what I mean by automatically is that eventually, the FreeIPA HA recovery will detect a crash of a node and bring up a replacement node.  Not a flow, more like a result of an action from a "cron'ed" status check.  This recovery is manual right now but will be automatic and this API call will be made to update the pillar config across an environment.  So there will be no thread-local user outside of one for the application itself, internal actor.  

As well its an internal to the Control Plane API call, not available external.  I think this is a difference in philosophy in parts of the application.  In general from a thunderhead standpoint only publicly available API calls check, usually by the API service that fronts the internal service.  Internal services are openly available to be used by any other microservice.  From a CB standpoint, we are checking every API call every time, no matter if it is public or not.  This makes it very hard for anything like an automated service to be able to make calls to CB based services.  It has no "user" to work with except for an Internal Actor which would have access to everything anyways.

Reply:
since anyone can make a cdp cli command or ui call using cb related apis, this is not true and every cb api can become public anytime, thus we enforce authorization, cc @lnardai

Reply:
The proper method for this seems to be the @InternalOnly annotation.

Reply:
Im not sure how "anyone" is able to do this.  This has to be a service deployed within the k8s CP or have an external API exposed via and API services for the CLI have to be defined in the CDP sdk and API documents.  This is the reason we have things like the environment service in the thunderhead fronting various things like the CB environment service or the datalake API service in thunderhead to front the CB datalake service.

What we are talking about is controlled access by internal trusted services within the control plane.  Are you saying there is some other way that we expose these APIs to the public?

Reply:
You are correct, the cdp-cli works through the API services. But if you check the UI many API calls are not relying on the API services but go through another route, called dps-gateway. For that reason we can't just leave API's "unprotected", we need to specify at least they should used only internally, otherwise with the right URL they are accessible.